conf/machine/include/rpi-default-providers.inc

```diff
@@ -7,6 +7,8 @@ PREFERRED_PROVIDER_virtual/libgles2 ?= "${@bb.utils.contains("MACHINE_FEATURES",
 PREFERRED_PROVIDER_virtual/libgl ?= "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "mesa", "mesa-gl", d)}"
 PREFERRED_PROVIDER_virtual/mesa ?= "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "mesa", "mesa-gl", d)}"
 PREFERRED_PROVIDER_virtual/libgbm ?= "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "mesa", "mesa-gl", d)}"
+PREFERRED_PROVIDER_vlc ?= "rpidistro-vlc"
+PREFERRED_PROVIDER_ffmpeg ?= "rpidistro-ffmpeg"
```

Comment:
Can we have the difference controlled via a knob, I think we should try to not override the provider unconditionally.

Reply:
These providers are only taken in through a machine configuration so having this default for them is not much different than having a kernel provider default as well (I'm obviously stretching the argument here for the sake of clarity). I would find that the same approach applies here.

Reply:
like how we control `vc4graphics` via MACHINE_FEATURES we need something like that here too.

Reply:
@kraj ^